app/src/main/java/uk/gov/onelogin/appcheck/AppIntegrityImpl.kt

```diff
@@ -1,26 +1,60 @@
 package uk.gov.onelogin.appcheck
 
 import android.util.Log
+import io.ktor.util.date.getTimeMillis
 import javax.inject.Inject
 import uk.gov.android.authentication.integrity.ClientAttestationManager
 import uk.gov.android.authentication.integrity.model.AttestationResponse
 import uk.gov.android.features.FeatureFlags
+import uk.gov.android.securestore.error.SecureStorageError
+import uk.gov.onelogin.appcheck.AppIntegrity.Companion.CLIENT_ATTESTATION
+import uk.gov.onelogin.appcheck.AppIntegrity.Companion.CLIENT_ATTESTATION_EXPIRY
+import uk.gov.onelogin.appcheck.AppIntegrity.Companion.SECURE_STORE_ERROR
 import uk.gov.onelogin.features.AppCheckFeatureFlag
+import uk.gov.onelogin.tokens.usecases.GetFromOpenSecureStore
+import uk.gov.onelogin.tokens.usecases.SaveToOpenSecureStore
 
 class AppIntegrityImpl @Inject constructor(
     private val featureFlags: FeatureFlags,
-    private val appCheck: ClientAttestationManager
+    private val appCheck: ClientAttestationManager,
+    private val saveToOpenSecureStore: SaveToOpenSecureStore,
+    private val getFromOpenSecureStore: GetFromOpenSecureStore
 ) : AppIntegrity {
     override suspend fun startCheck(): AppIntegrityResult {
-        return if (featureFlags[AppCheckFeatureFlag.ENABLED]) {
+        // This is just to not block from using the app as the Firebase Token expiry time is set for a longer time
```

Comment:
Maybe for now we just accept that we will call the AppCheck stuff every time we enter the app? Basically we just disregard the expiry time for now, it will remove some temporary complexity here

Reply:
I've added this because if you log out and try to log in you have to wait or delete the app completely from the device if you want to log in (within 15-20 minutes). But I can remove it